### ISSUE TITLE
[a11y]: OverflowMenu with the `enableV12Overflowmenu` shows accessibility violation on `aria-label`

### ISSUE THREAD

**Issue by makafsal** [carbon-design-system/carbon#19517]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.83.0

### React version

v18.3.0

### Automated testing tool and ruleset

IBM Equal accessibility checker

### Description

When I give an `aria-label` for the `OverflowMenu` component when `enableV12Overflowmenu` is enabled

```
<FeatureFlags enableV12Overflowmenu>
        <OverflowMenu aria-label="I'm causing the violation, I don't know why">
```

It shows a violation:

![Image](https://github.com/user-attachments/assets/c7e6195b-7fbc-44c5-9ffc-3fa4705216c4)

### Reproduction/example

https://stackblitz.com/edit/jh76k2jx?file=src%2FApp.jsx

### Steps to reproduce

1. Go to the [Stackblitz example](https://stackblitz.com/edit/jh76k2jx?file=src%2FApp.jsx)
2. Try to run the accessibility checker with and without `aria-label` for the `OverflowMenu` component. (Stay wrapped with `<FeatureFlags enableV12Overflowmenu>`)

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by annawen1** [comment]
Closing this issue as the documentation for the `enableV12Overflowmenu` shows to use `label` instead of `aria-label`: https://react.carbondesignsystem.com/?path=/docs/components-overflowmenu-feature-flag--flag-details#overview

Doing so resolves the a11y violation
